ext/nnstreamer/tensor_filter/tensor_filter_llama2.cc

```diff
@@ -0,0 +1,268 @@
+
+  /* mandatory methods */
+  tensor_filter_subplugin &getEmptyInstance ();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+
+  /* static methods */
+  static void init_filter_llama2c ();
+  static void fini_filter_llama2c ();
+
+  private:
+  static TensorFilterLlama2c *registered;
+  static const char *name;
+  static const accl_hw hw_list[];
+  static const int num_hw;
+
+  static Transformer *transformer;
+  static Tokenizer *tokenizer;
+  static Sampler *sampler;
+  static Config *config;
+  char *prompt; // prompt string
+  int steps; // number of steps to run for
+};
+
+void init_filter_llama2c (void) __attribute__ ((constructor));
+void fini_filter_llama2c (void) __attribute__ ((destructor));
+
+TensorFilterLlama2c *TensorFilterLlama2c::registered = nullptr;
+const char *TensorFilterLlama2c::name = "llama2c";
+Transformer *TensorFilterLlama2c::transformer = nullptr;
+Tokenizer *TensorFilterLlama2c::tokenizer = nullptr;
+Sampler *TensorFilterLlama2c::sampler = nullptr;
+Config *TensorFilterLlama2c::config = nullptr;
+
+/**
+ * @brief Construct a new llama2c subplugin instance
+ */
+TensorFilterLlama2c::TensorFilterLlama2c ()
+{
+}
+
+/**
+ * @brief Destructor of TensorFilterLlama2c
+ */
+TensorFilterLlama2c::~TensorFilterLlama2c ()
+{
+}
+
+/**
+ * @brief Method to get an empty object
+ */
+tensor_filter_subplugin &
+TensorFilterLlama2c::getEmptyInstance ()
+{
+  return *(new TensorFilterLlama2c ());
+}
+
+/**
+ * @brief Configure llama2c instance
+ */
+void
+TensorFilterLlama2c::configure_instance (const GstTensorFilterProperties *prop)
+{
+  // build the Transformer via the model file
+  build_transformer (transformer, (char *) prop->model_files[0]);
+  config = &transformer->config;
+
+  if (steps == 0 || steps > config->seq_len)
+    steps = config->seq_len; // override to ~max length
+
+  // build the Tokenizer via the tokenizer file
+  build_tokenizer (tokenizer, (char *) prop->model_files[1], config->vocab_size);
+
+  // build the Sampler
+  build_sampler (sampler, config->vocab_size, TEMPERATURE, TOPP, RNG_SEED);
+}
+
+/**
+ * @brief Invoke llama2c using input tensors
+ */
+void
+TensorFilterLlama2c::invoke (const GstTensorMemory *input, GstTensorMemory *output)
+{
+  char *result = NULL, *prompt;
+  int len;
+
+  prompt = (char *) input[0].data;
+  len = strlen (prompt);
+
+  if (len > 1) {
+    prompt[strlen (prompt) - 1] = ' '; // Remove new line from text file
+  } else {
+    prompt = NULL;
+  }
+
+  output->size = tokenizer->max_token_length * steps;
+
+  result = get_tokens (transformer, tokenizer, sampler, prompt, steps);
+
+  output[0].data = result;
+  output[0].size = strlen (result);
```

Comment:
L164 and L169 touch same var

Reply:
There was a mistake while processing output tensors.
Great thanks 👍